apps/modernization-ui/src/apps/search/laboratory-report/CriteriaFields.spec.tsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import { render, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { useForm } from 'react-hook-form';
+import { LabReportFilterEntry } from 'apps/search/laboratory-report/labReportFormTypes';
+import { CriteriaFields, LabReportCriteriaFields } from './CriteriaFields';
+import { AutocompleteSingleProps } from 'design-system/autocomplete';
+
+jest.mock('options/autocompete/ResultedTestsAutocomplete', () => ({
+    ResultedTestsAutocomplete: ({ id, label, onChange }: AutocompleteSingleProps) => (
+        <input
+            data-testid={id}
+            placeholder={label}
+            onChange={(e) => onChange?.({ name: e.target.value, value: e.target.value, label: e.target.value })}
+        />
+    )
+}));
+
+jest.mock('options/autocompete/CodedResultsAutocomplete', () => ({
+    CodedResultsAutocomplete: ({ id, label, onChange }: AutocompleteSingleProps) => (
+        <input
+            data-testid={id}
+            placeholder={label}
+            onChange={(e) => onChange?.({ name: e.target.value, value: e.target.value, label: e.target.value })}
+        />
+    )
+}));
+
+const LabReportCriteriaFieldsWithForm = () => {
+    const form = useForm<LabReportFilterEntry>({ defaultValues: {} });
+    return <LabReportCriteriaFields form={form} />;
+};
+
+const CriteriaFieldsWithForm = () => {
+    const form = useForm<LabReportFilterEntry>({ defaultValues: {} });
+    return <CriteriaFields form={form} />;
+};
+
+describe('LabReportCriteriaFields component', () => {
+    it('should render ResultedTestsAutocomplete with correct props', async () => {
+        const { getByTestId, getByPlaceholderText } = render(<LabReportCriteriaFieldsWithForm />);
```

Comment:
suggestion: The `getByTestId` should be used as a last resort [getByRole](https://testing-library.com/docs/queries/byrole) can be used here